terraform/moe_bucket.tf

```
@@ -8,3 +8,70 @@ resource "aws_s3_bucket" "moe_bucket" {
     prevent_destroy = true
   }
 }
+
+# IAM policy document that that allows read permissions on the MOE bucket.
+data "aws_iam_policy_document" "moe_bucket_read_doc" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:ListBucket",
+    ]
+
+    resources = [
+      aws_s3_bucket.moe_bucket.arn,
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:GetObject",
+    ]
+
+    resources = [
+      "${aws_s3_bucket.moe_bucket.arn}/*",
+    ]
+  }
+}
+
+# The policy that allows read access to the MOE S3 bucket
+resource "aws_iam_policy" "moe_bucket_read" {
+  name   = format("moe_bucket_read_%s", local.production_workspace ? "production" : terraform.workspace)
+  policy = data.aws_iam_policy_document.moe_bucket_read_doc.json
+}
+
+# IAM policy document that that allows write permissions on the MOE bucket.
+data "aws_iam_policy_document" "moe_bucket_write_doc" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:ListBucket",
+    ]
+
+    resources = [
+      aws_s3_bucket.moe_bucket.arn,
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:DeleteObject",
+      "s3:PutObject",
+    ]
+
+    resources = [
+      "${aws_s3_bucket.moe_bucket.arn}/*",
+    ]
```

Similar consolidation:
```suggestion
      "s3:DeleteObject",
      "s3:ListBucket",
      "s3:PutObject",
    ]

    resources = [
      aws_s3_bucket.moe_bucket.arn,
      "${aws_s3_bucket.moe_bucket.arn}/*",
    ]
```

---

I'm fine either way.  I can't decide which way I like better; both have their pluses and minuses.

---

See https://github.com/cisagov/cyhy_amis/pull/649#discussion_r1164469340